src/Workspaces/Remote/ServiceHub/Services/DesignerAttribute/RemoteDesignerAttributeIncrementalAnalyzer.cs

```diff
@@ -74,121 +90,82 @@ private async Task AnalyzeProjectAsync(Project project, Document? specificDocume
             // in this project.
             var projectVersion = await project.GetDependentSemanticVersionAsync(cancellationToken).ConfigureAwait(false);
 
-            var latestInfos = await ComputeLatestInfosAsync(
-                project, projectVersion, specificDocument, cancellationToken).ConfigureAwait(false);
-
             // Now get all the values that actually changed and notify VS about them. We don't need
             // to tell it about the ones that didn't change since that will have no effect on the
             // user experience.
-            //
-            //  !  is safe here as `i.changed` implies `i.info` is non-null.
-            var changedInfos = latestInfos.Where(i => i.changed).Select(i => i.data!.Value).ToList();
-            if (changedInfos.Count > 0)
+            var latestData = await ComputeLatestDataAsync(
+                project, specificDocument, projectVersion, cancellationToken).ConfigureAwait(false);
+
+            var changedData =
+                latestData.Where(d => !_documentToLastReportedInformation.TryGetValue(d.document.Id, out var existingInfo) || existingInfo.category != d.data.Category)
+                          .ToImmutableArray();
```

Comment:
If the user had a file that we classified, and then edits the code such that ComputeLatestDataAsync no longer returns anything for that file, do we need to report the now lack of data? And clear out the dictionary in that case?

Reply:
we do report the *lack* of data.  specifically, we report changes from whatever we last reported.  We don't actually clear out the dictionary because we want to store in teh dictionary the version of the last report, taht way we can ignore future events (that don't chanage hte state).  If we remove things from teh dictionary, that is akin to saying "we've never reported anything", and then we will just keep notifying VS of "it's not designable, it's not designable, it's not designable".

Reply:
Ah got it; all documents being queried still get added to the dictionary no matter what. Makes sense.